pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go

```diff
@@ -197,6 +199,11 @@ func (v *CloudStackDatacenterConfig) Validate() error {
 	}
 	azSet := make(map[string]bool)
 	for _, az := range v.Spec.AvailabilityZones {
+		errorMessages := validation.IsValidLabelValue(az.Name)
+		if len(errorMessages) > 0 {
```

Comment:
Why are we collecting a list of errorMessages here? Then we check for multiple errors when there can only be one a single one. It seems like it would make more sense to have

```
if azNameInvalid {
    return fmt.Errorf("invalid name)
}
```

Reply:
The error message from `errorMessages` can be multiple, here we collapse them into one. 
If we only use `invalid name`, customer misses the details about how to fix the error.

Reply:
could you change this for all the errors being returned by this function then? Otherwise it's only being used in one place.